### PR TITLE
fix: Updated the script to generate commits in the release

### PR DESCRIPTION
**Pull request by hjetpoluru**
## **Description**

This PR is to updated the existing script to generate commits in the release. I enhanced author attribution list, tagged labels with release version, and filtered out specific commit message like 'Changelog' or 'Merge pull request' or 'master-sync' .

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23914?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

Provide your github token in the development/generate-rc-commits.js
Provide the below command in the terminal and the cvs file is generated
node development/generate-rc-commits.js origin/Version-v11.14.0 origin/Version-v11.13.0

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: development/generate-rc-commits.js [hunk 1]
```diff
@@ -18,32 +28,39 @@ const authorTeams = {
...
     'Kate Johnson',
+    'Xiaoming Wang',
+    'Charly Chevalier',
+    'Mike B',
+  ],
+  'Extension UX': [
+    'David Walsh',
+    'Nidhi Kumari',
+    'Jony Bursztyn',
+    'Pedro Figueiredo',
```
Comment (pedronfigueiredo) [inline development/generate-rc-commits.js:39]: Confirmations UX

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by benjisclowder — APPROVED** [review]
Tested locally and got this csv file: 
image


Git command line in manual steps should be: node development/generate-rc-commits.js origin/Version-v11.15.0 origin/Version-v11.14.0.

**Review by benjisclowder — CHANGES_REQUESTED** [review]
Tested locally and got this csv file: 
image


Git command line in manual steps should be: node development/generate-rc-commits.js origin/Version-v11.15.0 origin/Version-v11.14.0

**Review by benjisclowder — APPROVED** [review]
LGTM.

**Review by chloeYue — APPROVED** [review]
LGTM !

**Pull request merged**